### ISSUE TITLE
Enhancement to read_database with a SQLAlchemy connection

### ISSUE THREAD

**Issue by matquant14** [pola-rs/polars#11668]
### Description

Love the new feature to enable parameterized queries using the read_database funciton.  I'd like to suggest some additional functionality when passing a SQLAlchemy connection.  Looking at the current implementation, it's converting the SQL using SQLAlchemy's Text class and then executing.  

Can additional functionality could be added, say by creating a flag in the execute_options parameter or adding another optional argument, whereby it skips that step of converting the string to a TextClause, and instead executes the query using the connection's exec_driver_sql method?  (This is what pandas does under the hood in it's read_sql function)

https://docs.sqlalchemy.org/en/20/core/connections.html#invoking-sql-strings-directly-to-the-driver

This should also allow parameters to be passed, and it permits qmark parameterization as well.  

This isn't an exact way to implement it, as I wrote it on the fly, but maybe add something in the ConnectionExecutor's execute method like:
```
def execute(
        self,
        query: str | Selectable,
        *,
        options: dict[str, Any] | None = None,
        dbapi_execution:bool=False,
        select_queries_only: bool = True,
    ) -> Self:
        """Execute a query and reference the result set."""
        if select_queries_only and isinstance(query, str):
            q = re.search(r"\w{3,}", re.sub(r"/\*(.|[\r\n])*?\*/", "", query))
            if (query_type := "" if not q else q.group(0)) in _INVALID_QUERY_TYPES:
                raise UnsuitableSQLError(
                    f"{query_type} statements are not valid 'read' queries"
                )
       
        if self.driver_name == "sqlalchemy"  and dbapi_execution:
            self.cursor.exec_driver_sql(query,**options)
            return self

        if self.driver_name == "sqlalchemy" and isinstance(query, str):
            from sqlalchemy.sql import text
        
            query = text(query)  # type: ignore[assignment]

        if (result := self.cursor.execute(query, **(options or {}))) is None:
            result = self.cursor  # some cursors execute in-place

        self.result = result
        return self

```

**Comment by mcrumiller** [comment]
I'm curious what the benefit of bypassing the Text class is?

**Comment by matquant14** [comment]
Not using the text class adds some flexibility, imho, when using qmark parameterization.  Of course it depends on your use case. 
 I've found, when using the text class, SQLAlchemy is pretty rigid in how it compiles and executes a raw SQL command.  The text class requires the parametrized SQL to be in a specific format, whereas a lot of RDBMSs accept qmark parameters by default.  

I work with a few different RDBMSs.  For me, there's less overhead in passing the same exact SQL, using qmark parameterization along w/ it's respective parameter vs. having to convert  SQL to use placeholders and pass parameters as dictionaries for one RDBMS and using a different format on the other.

**Comment by alexander-beedie** [comment]
> For me, there's less overhead in passing the same exact SQL, using qmark parameterization along w/ it's respective parameter vs. having to convert SQL to use placeholders and pass parameters as dictionaries for one RDBMS and using a different format on the other.

I can see the value in that; I've got a cleaner way to handle it though  ;)
PR incoming...

**Comment by alexander-beedie** [comment]
@matquant14: merged - will be available in `0.19.9` ✌️ 
(Added an example of this form of parameterisation to the docstring too).

**Comment by matquant14** [comment]
That's great!  Looking forward to it. Thanks for such a quick turnaround.
